builder/ghc-for-component-wrapper.nix

```
@@ -20,7 +20,7 @@ let
   docDir         = "$out/share/doc/ghc/html";
   # For musl we can use haddock from the buildGHC
   haddock        = if stdenv.hostPlatform.isLinux && stdenv.targetPlatform.isMusl
```

```suggestion
  haddock        = if stdenv.hostPlatform.isLinux && stdenv.targetPlatform.isMusl && !(stdenv.buildPlatform == stdenv.hostPlatform)
```
would work as well, instead of the nested ifs?

---

We should add a comment here though for the reasoning behind the convoluted if statement.

---

Yep, I'll fix it